pkg/ruler/manager.go

```diff
@@ -100,7 +111,62 @@ func NewDefaultMultiTenantManager(cfg Config, managerFactory ManagerFactory, reg
 		}, []string{"user"}),
 		registry: reg,
 		logger:   logger,
-	}, nil
+	}
+
+	var err error
+	switch cfg.AlertManagerDiscovery.Mode {
+	case alertmanagerdiscovery.ModeRing:
+		level.Info(logger).Log("msg", "using ring based alert manager discovery")
+		m.discoveryService, err = alertmanager.NewRing(cfg.AlertManagerRing, "ruler", m, logger, reg)
```

Comment:
[nit] Design wise looks a bit weird to me that the ring-based service discovery is in the `alertmanager` package, while the DNS-based one is in `alertmanagerdiscovery`. Have you considered moving `NewRing` to `alertmanagerdiscovery.NewRingProvider()`?

Reply:
Yea, I wasn't certain on where to put `NewRing` since I wanted to reuse the existing RingConfig, which would cause import cycles if I moved this to the alertmanagerdiscovery package, but I'll just move it and added a separate RingConfig there.